File: fathersDay/Dsa/Day 29 (14. Longest Common Prefix)/Solution.py
class Solution:
    def LongestCommonPrefix(self, strs: list[str]) -> str:
        if not strs:
            return ""
        prefix = strs[0]
        for s in strs[1:]:
            while not s.startswith(prefix):
                prefix = prefix[:-1]
                if not prefix:
                    return ""
        return prefix
# Example usage:
solution = Solution()
strs = ["flower", "flow", "flight"]
result = solution.LongestCommonPrefix(strs)
print(result)  # Output: "fl", as it is the longest common prefix among the strings

File: fathersDay/Dsa/Day 29 (14. Longest Common Prefix)/Solution.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

class Solution {
public:
    std::string LongestCommonPrefix(std::vector<std::string>& strs) {
        if (strs.empty()) {
            return "";
        }
        std::string prefix = strs[0];
        for (size_t i = 1; i < strs.size(); ++i) {
            while (strs[i].find(prefix) != 0) {
                prefix = prefix.substr(0, prefix.length() - 1);
                if (prefix.empty()) {
                    return "";
                }
            }
        }
        return prefix;
    }
};

int main() {
    Solution solution;
    std::vector<std::string> strs = {"flower", "flow", "flight"};
    std::string result = solution.LongestCommonPrefix(strs);
    std::cout << '"' << result << '"' << std::endl; // Output: "fl"
    return 0;
}
